src/Microsoft.AspNetCore.Cors/Infrastructure/CorsService.cs

```diff
@@ -136,12 +136,21 @@ private static void PopulateResult(HttpContext context, CorsPolicy policy, CorsR
             result.SupportsCredentials = policy.SupportsCredentials;
             result.PreflightMaxAge = policy.PreflightMaxAge;
 
-
+            // https://fetch.spec.whatwg.org/#http-new-header-syntax
             AddHeaderValues(result.AllowedExposedHeaders, policy.ExposedHeaders);
-            AddHeaderValues(result.AllowedMethods, policy.Methods);
-            AddHeaderValues(result.AllowedHeaders, policy.Headers);
+            AddHeaderValues(result.AllowedMethods, policy.AllowAnyMethod && policy.SupportsCredentials ? GetCorsMethod() : policy.Methods);
+            AddHeaderValues(result.AllowedHeaders, policy.AllowAnyHeader && policy.SupportsCredentials ? GetCorsRequestHeaders() : policy.Headers);
+
+            IList<string> GetCorsMethod() =>
```

Comment:
Can we do this inline? This doesn't need to be a function.

Reply:
Why don't you like the function?
I don't like having an uber epic line above (line 141-142) (makes it way harder to read) and there are two options here.
1) local variable
2) local function

of those two the local function is the best IMO as:
1) It names the operation
2) It hides the implementation detail later in the method so that you can read the body top-down.

I'm only saying this to make you think about it. I've humored you and changed to compute the temp in a local variable.

If you are concerned about the perf:
1) premature optimization
2) the JIT will most likely inline it on the generated code

Reply:
Isn't this easier to read:

```C#
if (policy.AllowAnyHeader && policy.SupportsCredentials && result.IsPreflightRequest)
{
   result.AllowedHeaders.Add(context.Request.Headers[CorsConstants.AccessControlRequestMethod]);
}
```